src/core/utils/Popover.tsx

```diff
@@ -85,6 +85,9 @@ export const Popover = React.forwardRef((props: PopoverProps, ref) => {
 
 export const hideOnEsc = {
   fn(instance: Instance) {
+    // will always refer to the correct document instance (e.g. in a popout window)
+    const _document = instance.reference?.ownerDocument ?? document;
```

Comment:
Maybe we need to add a util like this: https://github.com/mui-org/material-ui/blob/next/packages/material-ui-utils/src/ownerDocument.ts